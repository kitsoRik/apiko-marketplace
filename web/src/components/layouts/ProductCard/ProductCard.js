import React from 'react';

import Black from '../../../assets/icons/black.png';
import "./ProductCard.scss";
import Icon from '../Icon';
import ModalLoading from '../ModalLoading/ModalLoading';
import HeartIcon from '../../icons/HeartIcon';

const ProductCard = ({ className, id, title, price, iconName, changingSaveState, saved = false, onSavedChange = () => { }, ...props }) => {
    return (
        <div className={`product-card ${className ?? ""}`} {...props}>
            <div className="product-card-icon">
                <img src={Black} />
            </div>
            <div className="product-card-info">
                <span className="product-card-info-name">{title}</span>
                <span className="product-card-info-price">{price}</span>
            </div>
            <div className="product-card-like">
                <HeartIcon filed={saved} onClick={() => onSavedChange(!saved)} />
                {changingSaveState && <ModalLoading style={{ padding: "6px" }} />}
            </div>
        </div>
    );
}

export default ProductCard;