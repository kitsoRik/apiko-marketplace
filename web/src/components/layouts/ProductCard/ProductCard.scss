.product-card {
    padding: 4px;

    width: 100%;
    height: 100%;

    position: relative;

    display: grid;
    grid-template-rows: 150px auto;

    background-color: #fff;

    border-radius: 4px;
    border: 1px solid #e4e4e4;
    box-sizing: border-box;

    box-shadow: 0px 4px 14px rgba(121, 121, 121, 0.0527344);

    &-icon {
        overflow: hidden;
        display: flex;
        justify-content: center;
        align-items: center;
        & > img {
            height: 100%;
        }
    }

    &-info {
        display: flex;
        flex-direction: column;

        &-title {
            padding: 6px 12px 0;

            color: black;
            text-decoration: none;

            &:hover {
                text-decoration: underline;
            }
        }

        &-price {
            padding: 7px 12px 10px;

            font-weight: bold;

            &::before {
                content: "$";
            }
        }
    }

    &-like {
        height: 30px;
        width: 30px;

        top: 133px;
        right: 7px;

        position: absolute;

        display: flex;
        justify-content: center;
        align-items: center;

        background-color: white;

        border-radius: 50%;

        overflow: hidden;

        cursor: pointer;
    }
}
