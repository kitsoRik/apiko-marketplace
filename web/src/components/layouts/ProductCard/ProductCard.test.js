import React from 'react';
import { create } from 'react-test-renderer';
import ProductCard from '.';


describe("Label", () => {
    it("should be rendered", () => {
        const component = create(
            <ProductCard
                className="my-custom-class-name"
                id={0}
                title="My custom title"
                price={55.07}
                iconName={"http://..."} />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with saved state", () => {
        const component = create(
            <ProductCard
                className="my-custom-class-name"
                id={0}
                title="My custom title"
                price={55.07}
                saved={true}
                iconName={"http://..."} />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with changing saved state", () => {
        const component = create(
            <ProductCard
                className="my-custom-class-name"
                id={0}
                title="My custom title"
                price={55.07}
                saved={true}
                changingSaveState={true}
                iconName={"http://..."} />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});