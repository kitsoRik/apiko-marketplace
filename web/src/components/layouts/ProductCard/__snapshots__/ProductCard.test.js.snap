// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should be rendered 1`] = `
<div
  className="product-card my-custom-class-name"
>
  <div
    className="product-card-icon"
  >
    <img
      alt="Product icon"
      src="black.png"
    />
  </div>
  <div
    className="product-card-info"
  >
    <span
      className="product-card-info-name"
    >
      My custom title
    </span>
    <span
      className="product-card-info-price"
    >
      55.07
    </span>
  </div>
  <div
    className="product-card-like"
  >
    <svg
      className="heart-icon "
      onClick={[Function]}
      viewBox="0 0 512 512"
      x="0px"
      y="0px"
    >
      <g>
        <path
          d="M256,468.8c-7.9,0-15.3-3-20.8-8.4L55.6,284.3c-27.2-26.7-42.2-62.2-42.2-100c0-37.8,15-73.3,42.2-100
 c27.2-26.7,63.5-41.4,102.2-41.4c33.9,0,66.9,11.7,92.7,32.9l5.4,4.4l5.4-4.4c25.8-21.2,58.8-32.9,92.7-32.9
 c38.7,0,75,14.7,102.2,41.4c27.2,26.7,42.2,62.2,42.2,100c0,37.8-15,73.3-42.2,100L276.8,460.4C271.2,465.8,263.8,468.8,256,468.8z"
          fill="none"
        />
        <path
          d="M354.1,51.3c36.4,0,70.6,13.9,96.3,39c25.6,25.1,39.7,58.5,39.7,94s-14.1,68.9-39.7,94L270.8,454.3c-3.9,3.9-9.2,6-14.9,6
 c-5.6,0-10.9-2.1-14.9-6L61.6,278.2c-25.6-25.1-39.7-58.5-39.7-94c0-35.5,14.1-68.9,39.7-94c25.6-25.2,59.8-39,96.3-39
 c32,0,63,11,87.3,31l10.8,8.9l10.8-8.9C291.1,62.3,322.1,51.3,354.1,51.3 M354.1,34.3c-34.9,0-69.8,11.6-98.1,34.9
 c-28.3-23.2-63.2-34.8-98.1-34.8c-39.2,0-78.3,14.6-108.2,43.9c-28.9,28.3-44.8,66-44.8,106.1S20.8,262,49.7,290.4l179.5,176.1
 c7.4,7.3,17.1,10.9,26.8,10.9s19.4-3.6,26.8-10.9l179.5-176.1c28.9-28.3,44.8-66,44.8-106.1c0-40.1-15.9-77.8-44.8-106.1
 C432.4,48.9,393.2,34.3,354.1,34.3L354.1,34.3z"
          fill="#B7B7B7"
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`Label should render with changing saved state 1`] = `
<div
  className="product-card my-custom-class-name"
>
  <div
    className="product-card-icon"
  >
    <img
      alt="Product icon"
      src="black.png"
    />
  </div>
  <div
    className="product-card-info"
  >
    <span
      className="product-card-info-name"
    >
      My custom title
    </span>
    <span
      className="product-card-info-price"
    >
      55.07
    </span>
  </div>
  <div
    className="product-card-like"
  >
    <svg
      className="heart-icon "
      onClick={[Function]}
      viewBox="0 0 512 512"
      x="0px"
      y="0px"
    >
      <g>
        <path
          d="M256,468.8c-7.9,0-15.3-3-20.8-8.4L55.6,284.3c-27.2-26.7-42.2-62.2-42.2-100c0-37.8,15-73.3,42.2-100
 c27.2-26.7,63.5-41.4,102.2-41.4c33.9,0,66.9,11.7,92.7,32.9l5.4,4.4l5.4-4.4c25.8-21.2,58.8-32.9,92.7-32.9
 c38.7,0,75,14.7,102.2,41.4c27.2,26.7,42.2,62.2,42.2,100c0,37.8-15,73.3-42.2,100L276.8,460.4C271.2,465.8,263.8,468.8,256,468.8z"
          fill="#349A89"
        />
        <path
          d="M354.1,51.3c36.4,0,70.6,13.9,96.3,39c25.6,25.1,39.7,58.5,39.7,94s-14.1,68.9-39.7,94L270.8,454.3c-3.9,3.9-9.2,6-14.9,6
 c-5.6,0-10.9-2.1-14.9-6L61.6,278.2c-25.6-25.1-39.7-58.5-39.7-94c0-35.5,14.1-68.9,39.7-94c25.6-25.2,59.8-39,96.3-39
 c32,0,63,11,87.3,31l10.8,8.9l10.8-8.9C291.1,62.3,322.1,51.3,354.1,51.3 M354.1,34.3c-34.9,0-69.8,11.6-98.1,34.9
 c-28.3-23.2-63.2-34.8-98.1-34.8c-39.2,0-78.3,14.6-108.2,43.9c-28.9,28.3-44.8,66-44.8,106.1S20.8,262,49.7,290.4l179.5,176.1
 c7.4,7.3,17.1,10.9,26.8,10.9s19.4-3.6,26.8-10.9l179.5-176.1c28.9-28.3,44.8-66,44.8-106.1c0-40.1-15.9-77.8-44.8-106.1
 C432.4,48.9,393.2,34.3,354.1,34.3L354.1,34.3z"
          fill="none"
        />
      </g>
    </svg>
    <div
      className="modal-loading undefined"
      darken="true"
      onClick={[Function]}
      style={
        Object {
          "padding": "6px",
        }
      }
    >
      <img
        alt="Loading"
        src="loading.svg"
      />
    </div>
  </div>
</div>
`;

exports[`Label should render with saved state 1`] = `
<div
  className="product-card my-custom-class-name"
>
  <div
    className="product-card-icon"
  >
    <img
      alt="Product icon"
      src="black.png"
    />
  </div>
  <div
    className="product-card-info"
  >
    <span
      className="product-card-info-name"
    >
      My custom title
    </span>
    <span
      className="product-card-info-price"
    >
      55.07
    </span>
  </div>
  <div
    className="product-card-like"
  >
    <svg
      className="heart-icon "
      onClick={[Function]}
      viewBox="0 0 512 512"
      x="0px"
      y="0px"
    >
      <g>
        <path
          d="M256,468.8c-7.9,0-15.3-3-20.8-8.4L55.6,284.3c-27.2-26.7-42.2-62.2-42.2-100c0-37.8,15-73.3,42.2-100
 c27.2-26.7,63.5-41.4,102.2-41.4c33.9,0,66.9,11.7,92.7,32.9l5.4,4.4l5.4-4.4c25.8-21.2,58.8-32.9,92.7-32.9
 c38.7,0,75,14.7,102.2,41.4c27.2,26.7,42.2,62.2,42.2,100c0,37.8-15,73.3-42.2,100L276.8,460.4C271.2,465.8,263.8,468.8,256,468.8z"
          fill="#349A89"
        />
        <path
          d="M354.1,51.3c36.4,0,70.6,13.9,96.3,39c25.6,25.1,39.7,58.5,39.7,94s-14.1,68.9-39.7,94L270.8,454.3c-3.9,3.9-9.2,6-14.9,6
 c-5.6,0-10.9-2.1-14.9-6L61.6,278.2c-25.6-25.1-39.7-58.5-39.7-94c0-35.5,14.1-68.9,39.7-94c25.6-25.2,59.8-39,96.3-39
 c32,0,63,11,87.3,31l10.8,8.9l10.8-8.9C291.1,62.3,322.1,51.3,354.1,51.3 M354.1,34.3c-34.9,0-69.8,11.6-98.1,34.9
 c-28.3-23.2-63.2-34.8-98.1-34.8c-39.2,0-78.3,14.6-108.2,43.9c-28.9,28.3-44.8,66-44.8,106.1S20.8,262,49.7,290.4l179.5,176.1
 c7.4,7.3,17.1,10.9,26.8,10.9s19.4-3.6,26.8-10.9l179.5-176.1c28.9-28.3,44.8-66,44.8-106.1c0-40.1-15.9-77.8-44.8-106.1
 C432.4,48.9,393.2,34.3,354.1,34.3L354.1,34.3z"
          fill="none"
        />
      </g>
    </svg>
  </div>
</div>
`;
